src/Aspire.Hosting/ApplicationModel/ReplicaInstancesAnnotation.cs

```diff
@@ -6,9 +6,11 @@
 namespace Aspire.Hosting.ApplicationModel;
 
 /// <summary>
-/// Keep track of active replicas ids from DCP.
+/// Keep track of instances from DCP.
 /// </summary>
 internal class ReplicaInstancesAnnotation : IResourceAnnotation
 {
-    public ConcurrentDictionary<string, string> Instances { get; } = [];
+    public ConcurrentDictionary<string, Instance> Instances { get; } = [];
```

Comment:
Is there any benefit to this being a dictionary? Seems like it could be an `ImmutableArray<Instance>`